src/mlpack/core/data/detect_file_type_impl.hpp

```diff
@@ -252,7 +300,7 @@ inline FileType AutoDetectFile(std::fstream& stream,
  * @param filename Name of the file whose type we should detect.
  * @return Detected type of file.
```

Comment:
This documentation is now incorrect: the detected type is not returned, but instead `opts` is modified directly.  Can you update this for clarity?